### ISSUE TITLE
WARNING:tensorflow:tf.keras.mixed_precision.experimental.LossScaleOptimizer

### ISSUE THREAD

**Issue by innat** [tensorflow/tensorflow#47956]
In my training setup, I've customized the `.fit` method by overriding the `train_step` function. 

```
class CustomFit(tf.keras.Model()):
	def train_step():
    	# do some modification 

    def test_step():
		# no modification

model = CustomFit()
model.compile(loss, optimizer, metrics)
model.fit()
```

Training start at this point as it should be but a warning message appears, though until now I didn't find any other issue except this annoying warning message. 

> WARNING:tensorflow:tf.keras.mixed_precision.experimental.LossScaleOptimizer is deprecated. Please use tf.keras.mixed_precision.LossScaleOptimizer instead. Note that the non-experimental LossScaleOptimizer does not take a DynamicLossScale but instead takes the dynamic configuration directly in the constructor. For example:
  opt = tf.keras.mixed_precision.experimental.LossScaleOptimizer(opt)

In my code, by myself, I didn't write anywhere this statement `opt = tf.keras.mixed_precision.experimental.LossScaleOptimizer(opt)`. What is this all about? Do I need to pass optimizer like this way before setting in `.compile`? Like 

```
opt = Adam()
opt = tf.keras.mixed_precision.experimental.LossScaleOptimizer(opt)
model.compile(loss, metrics, optimizer=opt)
```
Or, I can just ignore this warning? When I should use it and not?

**Comment by Saduf2019** [comment]
@innat 
Please refer to [this comment](https://github.com/tensorflow/tensorflow/issues/46589#issuecomment-774097322) and let us know after you upgrade.

**Comment by innat** [comment]
I saw, and read. What's the catch?

**Comment by innat** [comment]
Let me clarify. I didn't face any `ERROR`. The model runs normally but at the beginning, it's showing this warning message. So, I suspect there something I need to concern about on which `tf` is a warning.

**Comment by Saduf2019** [comment]
@innat 
The warning msg does not effect the model performance, can you share your system information , tf version cuda version, the links shared was for you to hide the warning as suggested in the link and resolved by many users in the same issue.

**Comment by innat** [comment]
Hiding some non-essential log message makes sense. But this warning tells something very specifically. 

> WARNING:tensorflow:tf.keras.mixed_precision.experimental.LossScaleOptimizer is deprecated. Please use tf.keras.mixed_precision.LossScaleOptimizer instead. Note that the non-experimental LossScaleOptimizer does not take a DynamicLossScale but instead takes the dynamic configuration directly in the constructor. For example:
opt = tf.keras.mixed_precision.experimental.LossScaleOptimizer(opt)

Like, somewhere in my code I used this `experimental.LossScaleOptimizer` and it warning me not to use the experimental version. So, I suspected somehow `tf` in the backend, use it for loss scaling. My `tf` version `2.4.0`, CUDA: `11.0`, cuDNN: `8.0.4`
